WordPress/Classes/ViewRelated/Post/Prepublishing/PrepublishingViewController.swift

```diff
@@ -170,14 +168,23 @@ final class PrepublishingViewController: UIViewController, UITableViewDataSource
 
     private func setupPublishButton() -> UIView {
         let footerView = UIView()
-        footerView.addSubview(publishButton)
-        publishButton.translatesAutoresizingMaskIntoConstraints = false
-        footerView.pinSubviewToSafeArea(publishButton, insets: Constants.nuxButtonInsets)
 
-        publishButton.addTarget(self, action: #selector(publish), for: .touchUpInside)
+        let hostingViewController = UIHostingController(rootView: PublishButton(viewModel: publishButtonViewModel).tint(Color(uiColor: .primary)))
+        addChild(hostingViewController)
+
+        footerView.addSubview(hostingViewController.view)
+        hostingViewController.view.translatesAutoresizingMaskIntoConstraints = false
+        footerView.pinSubviewToSafeArea(hostingViewController.view, insets: Constants.nuxButtonInsets)
 
         updatePublishButtonLabel()
 
+        if FeatureFlag.offlineMode.enabled {
+            updatePublishButtonState()
+            mediaPollingTimer = Timer.scheduledTimer(withTimeInterval: 0.1, repeats: true) { [weak self] _ in
```

Comment:
This is not particularly elegant, but I don't think there is a simple and reliably way to use `MediaCoordinator` and its callbacks to achieve the same thing. This, on the hand, is bulletproof.